admin_site/system/models.py

```diff
@@ -203,22 +199,17 @@ def update_package_status(self, name, do_add):
     def __str__(self):
         return self.name
 
-    def __unicode__(self):
-        return self.name
-
 
 class PackageInstallInfo(models.Model):
     do_add = models.BooleanField(default=True)
-    package = models.ForeignKey(Package)
+    package = models.ForeignKey(Package, on_delete=models.CASCADE)
     custom_packages = models.ForeignKey(CustomPackages,
-                                        related_name='install_infos')
+                                        related_name='install_infos',
+                                        on_delete=models.PROTECT)
```

Comment:
I've only ever seen `models.CASCADE` used -- what does `models.PROTECT` do? (Will `PackageInstallInfo` survive the deletion of the related `CustomPackages` object?)

Reply:
`models.PROTECT` refuses to let you delete the object on the other side of the link without unsetting the link first.

The third option is `SET_NULL`

But in this case, the system will refuse to delete the `CustomPackages` object as long as the `PackageInstallInfo` is linked to it. If that is not what we want, it can be changed.

Reply:
That behaviour sounds unproblematic. (I just thought I'd check.)